graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/UpdateModelBuilder.java

```diff
@@ -43,37 +44,74 @@ MutationModel build() throws SkipException {
+    Optional<Directive> cqlUpdateDirective = DirectiveHelper.getDirective("cql_update", operation);
+    boolean ifExists = computeIfExists(cqlUpdateDirective);
 
     ReturnType returnType = getReturnType("Mutation " + operationName);
-    if (returnType != SimpleReturnType.BOOLEAN) {
-      invalidMapping("Mutation %s: updates can only return Boolean", operationName);
+    if (returnType != SimpleReturnType.BOOLEAN && !(returnType instanceof ResponsePayloadModel)) {
+      invalidMapping(
+          "Mutation %s: invalid return type. Expected Boolean or a response payload",
+          operationName);
       throw SkipException.INSTANCE;
     }
 
-    List<InputValueDefinition> inputs = operation.getInputValueDefinitions();
-    if (inputs.isEmpty()) {
+    if (returnType instanceof ResponsePayloadModel) {
+      ResponsePayloadModel payload = (ResponsePayloadModel) returnType;
+      Set<String> unsupportedFields =
+          payload.getTechnicalFields().stream()
+              .filter(f -> f != ResponsePayloadModel.TechnicalField.APPLIED)
+              .map(ResponsePayloadModel.TechnicalField::getGraphqlName)
+              .collect(Collectors.toCollection(HashSet::new));
+      payload.getEntityField().ifPresent(e -> unsupportedFields.add(e.getName()));
+      if (!unsupportedFields.isEmpty()) {
+        warn(
+            "Mutation %s: 'applied' is the only supported field in update response payloads. Others will always be null (%s).",
+            operationName, String.join(", ", unsupportedFields));
+      }
+    }
+
+    List<InputValueDefinition> arguments = operation.getInputValueDefinitions();
+    if (arguments.isEmpty()) {
       invalidMapping(
-          "Mutation %s: updates must take the entity input type as the first argument",
+          "Mutation %s: updates must take either the entity input type or a list of primary key fields",
           operationName);
       throw SkipException.INSTANCE;
     }
 
-    if (inputs.size() > 1) {
-      invalidMapping("Mutation %s: updates can't have more than one argument", operationName);
-      throw SkipException.INSTANCE;
+    EntityModel entity;
+    InputValueDefinition firstArgument = arguments.get(0);
+    Optional<EntityModel> entityFromFirstArgument = findEntity(firstArgument);
+    Optional<String> entityArgumentName =
+        entityFromFirstArgument.map(__ -> firstArgument.getName());
+    List<ConditionModel> ifConditions;
+    if (entityFromFirstArgument.isPresent()) {
+      if (arguments.size() > 1) {
+        invalidMapping(
+            "Mutation %s: if an update takes an entity input type, it must be the only argument",
+            operationName);
+        throw SkipException.INSTANCE;
+      }
+      entity = entityFromFirstArgument.get();
+      ifConditions = Collections.emptyList();
+    } else {
+      entity = entityFromDirective(cqlUpdateDirective, "update", "cql_update");
+      ConditionsModelBuilder.Conditions conditions = buildOnlyIfConditions(entity);
```

Comment:
The problem is that if we build the where conditions based on the current `buildConditions` method, it is trying to add the to WHERE clause for all fields not annotated with `@cql_if`. So for example, when we have this query:
```
    + "  updateUserEQCustomPayload(\n"
            + "    pk: Int\n"
            + "    username: String\n"
            + "    age: Int @cql_if(field: \"age\", predicate: EQ)\n"
```
It will try to add both `pk` and `username` to WHERE conditions, however, this is not correct. We want to have only the PK in the `where` conditions, age in the `IF` clause, and `username` needs to be updated.
The proposed logic (similar to `DeleteModelBuilder`) will put `username` in the IF clause, and it will not update the value as expected.
(It will also fail with:
` non-primary key argument username must be indexed in order to use it in a condition`
)

If we want to keep the signature of the update as is (without requiring the `cql_where` on each field used in the WHERE) we need to use `getPrimaryKeyWhereConditions`. 
Otherwise, we have two options:
1. Require `cql_where` on all fields that will be used in the `WHERE`clause. If the field is not annotated with the `cql_where`, we are not adding it to a WHERE clause.
2. Use the existing `getPrimaryKeyWhereConditions` to generate WHERE clause + all fields annotated with `cql_where`. If the field is present in both `getPrimaryKeyWhereConditions` and also annotated with `cql_where`, we are overriding its predicate (i.e.: EQ`, `IN`) with the predicate supplied in the `cql_where`

I think that the latter approach is more user-friendly because it does not require `cql_where` directive for all PK fields.
@olim7t Wdyt?
Lastly, as this PR is about IF conditions and already adds more logic (IF EXISTS, returning payload) maybe we should handle `cql_where` in a separate PR (and ticket)?